cmd/snap/cmd_report_problem.go

```diff
@@ -0,0 +1,214 @@
+}
+
+func collectLinks(linksBag map[string]map[string]bool, fromSnap *client.Snap) {
+	for category, links := range fromSnap.Links {
+		m := linksBag[category]
+
+		if m == nil {
+			m = make(map[string]bool, len(links))
+		}
+
+		for _, link := range links {
+			logger.Debugf("category %v link %v", category, link)
+			if link == "" {
+				continue
+			}
+			if skipGenericLink(category, link) {
+				continue
+			}
+			m[link] = true
+		}
+
+		if len(m) > 0 {
+			linksBag[category] = m
+		}
+	}
+}
+
+func addCategoryLinks(linksBag map[string]map[string]bool, category, link string) {
+	if link == "" {
+		return
+	}
+
+	if skipGenericLink(category, link) {
+		return
+	}
+
+	c := linksBag[category]
+	if c == nil {
+		c = make(map[string]bool)
+	}
+
+	c[link] = true
+	linksBag[category] = c
+}
+
+// TODO:GOVERSION: use maps.Keys()
+func keys[K comparable, V any](m map[K]V) []K {
+	keys := make([]K, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	return keys
+}
+
+func (x *reportProblemCmd) reportOne(snapName string) bool {
+	remoteSnap, _, remoteErr := x.client.FindOne(snap.InstanceSnap(snapName))
+	localSnap, _, localErr := x.client.Snap(snapName)
+
+	allLinks := map[string]map[string]bool{}
+
+	if remoteErr != nil {
+		logger.Debugf("remote snap err: %v\n", remoteErr)
+	}
+	if remoteSnap != nil {
+		collectLinks(allLinks, remoteSnap)
+		addCategoryLinks(allLinks, "contact", remoteSnap.Contact)
+		addCategoryLinks(allLinks, "website", remoteSnap.Website)
+	}
+
+	if localErr != nil {
+		logger.Debugf("local snap err: %v\n", localErr)
+	}
+	if localSnap != nil {
+		collectLinks(allLinks, localSnap)
+		addCategoryLinks(allLinks, "contact", localSnap.Contact)
+		addCategoryLinks(allLinks, "website", localSnap.Website)
+	}
+
+	logger.Debugf("all links: %+v", allLinks)
+
+	if len(allLinks) == 0 {
+		fmt.Fprintf(Stdout, i18n.G("Publisher of snap %q has not listed any points of contact.\n"), snapName)
+		return false
+	}
+
+	fmt.Fprintf(Stdout, i18n.G("Publisher of snap %q has listed the following points of contact:\n"), snapName)
+
+	w := tabwriter.NewWriter(Stdout, 2, 2, 1, ' ', 0)
+
+	if c, ok := allLinks["contact"]; ok {
+		c := keys(c)
+		sort.Strings(c)
+		fmt.Fprintf(w, i18n.G("\tContact:\n"))
+		for _, k := range c {
+			fmt.Fprintf(w, "\t\t%s\n", k)
+		}
+	}
+
+	if c, ok := allLinks["issues"]; ok {
+		c := keys(c)
+		sort.Strings(c)
+		fmt.Fprintf(w, i18n.G("\tIssue reporting:\n"))
+		for _, k := range c {
+			fmt.Fprintf(w, "\t\t%s\n", k)
+		}
+	}
+
+	if c, ok := allLinks["website"]; ok {
+		c := keys(c)
+		sort.Strings(c)
+		fmt.Fprintf(w, i18n.G("\tWebsite:\n"))
+		for _, k := range c {
+			fmt.Fprintf(w, "\t\t%s\n", k)
+		}
+	}
+
+	for _, sourceName := range []string{"source", "source-code"} {
+		if c, ok := allLinks[sourceName]; ok {
+			c := keys(c)
+			sort.Strings(c)
+			fmt.Fprintf(w, i18n.G("\tSource code:\n"))
+			for _, k := range c {
+				fmt.Fprintf(w, "\t\t%s\n", k)
+			}
+			break
+		}
+	}
+	w.Flush()
```

Comment:
Discussing with @degville we ended up with an idea that it'd be nice to offer the user an option to open a browser pointing to the issue tracker link, something like:

```
Would you like to open issue tracker link in browser? [Y/n]
```

iff the user is in an interactive session, which looks like a desktop and `xdg-open` is available on the $PATH.